src/lib/hasFieldSelection.ts

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable require-yield */
+
+import {
+  FieldNode,
+  visit,
+  FragmentDefinitionNode,
+  BREAK,
+  GraphQLResolveInfo,
+} from "graphql"
+
+export const hasFieldSelection = (
+  resolveInfo: GraphQLResolveInfo,
+  match: (nodeName: string) => boolean
+): boolean => {
+  if (!resolveInfo.fieldNodes) return false
+  const firstFieldNode = resolveInfo.fieldNodes[0]
```

Comment:
Perhaps leave a TODO that us only reflecting on the first field node may be suspect to breaking in edge cases.